### ISSUE TITLE
Working with assertDatabaseHas on Sqlite db failing at random asserts

### ISSUE THREAD

**Issue by webmake** [laravel/framework#19391]
- Laravel Version: 5.4
- Database Driver & Version:
sqlite 3.8.2

### Description:
Working with floats (no matter float, double, decimal in schema) and asserting that values exists, there is uncomfortable to work with `assertDatabaseHas` method, because it doesn't find some magic numbers, and fails at random, even value actually exists.
I have some those **magic numbers**, but seems no one still hadn't such situation, it might be sqlite bug, but I suggest to change operator `=` to `like`.

### Steps To Reproduce:
schema:
```
        Schema::create('some', function(Blueprint $table) {
            $table->increments('id');
            $table->double('commission', 11, 3);
            $table->timestamps();
        });
```
phpunit test:
```
        factory(Some::class)->create([
            'commission' => 2,
        ]);
        \DB::update('update some set commission = `commission` + ' . 0.28);
        $this->assertDatabaseHas('some', [
            'commission' => 2.28,
        ]);
```
gives:
```
Failed asserting that a row in the table [some] matches the attributes {
    "commission": "2.28"
}.

Found: [
    {
        "id": "1",
        "commission": "2.28",
        "created_at": "2017-05-29 11:11:38",
        "updated_at": "2017-05-29 11:11:38"
    }
].
```

Simplified model in sqlite, to see whats happening:
```
create table "some" (
"id" integer not null primary key autoincrement,
"commission" float not null,
"created_at" datetime null,
"updated_at" datetime null
);
insert into "some" ("commission", "updated_at", "created_at") values (2, '2017-05-29 11:15:32', '2017-05-29 11:15:32');
update some set commission = `commission` + 0.28;
```
So executed test assert looks like:
```
sqlite> select count(*) as aggregate from "some" where ("commission" = 2.28);
0
```
Suggestion to change `=` sign to `like`, **if there is possibility**, like SQLiteGrammar does. Then result would be:
```
sqlite> select count(*) as aggregate from "some" where ("commission" like 2.28);
1
```

**Comment by sisve** [comment]
No, the solution isn't to switch to string comparison operators. This is a common theme when working with floating point numbers, they are rarely exactly the value you expect. Based on the blog post [The PHP floating point precision is wrong by default](https://www.leaseweb.com/labs/2013/06/the-php-floating-point-precision-is-wrong-by-default/), could you try and set `ini_set('precision', 17);` in your test to see if the output changes?

It is highly probable that the actual value somewhere is 2.2800000000000002 or similarly close to 2.28, but not exactly equal to that value. That's why assertEquals has a $delta to allow small differences.

> assertEquals(float $expected, float $actual[, string $message = '', float $delta = 0])
> Reports an error identified by $message if the two floats $expected and $actual are not within $delta of each other.
> Please read "What Every Computer Scientist Should Know About Floating-Point Arithmetic" to understand why $delta is neccessary.

Source: [PHPUnit Manual - Appendix A. Assertions, assertEquals](https://phpunit.de/manual/current/en/appendixes.assertions.html)
Ref: [https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html](https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html)

**Comment by webmake** [comment]
Actually, it doesn't matter, giving string or float, `assertDatabaseHas` still returns that not found. Maybe something similar like `$delta` at  `assertDatabaseHas` would help? Actually working test **cheat** look like:
```
$actual = Some::first();
$this->assertEquals(2.28 , $actual->commission);
```
But for each value checking it is not a solution, while I could trust in database search

**Comment by themsaid** [comment]
Closing for lack of activity and it doesn't really seem to be an issue in the framework, feel free to ping me if you think otherwise though.

**Comment by ChrisReid** [comment]
@almokhtarbr as he said, it's not an issue in the framework. Should issues be left open indefinitely?

**Comment by arvins-wittymanager** [comment]
I'm having the same issue as @webmake and upon inspecting the debugger, the expected and actual values are visually the same `{float} 1.61` but the test still fails.

A little background, the original value is 1.6071, which is saved in a `$table->float('tax', 10)->default(0.00);` migration file, which returns type `double (10, 2)` as shown in the [grammar file](https://github.com/laravel/framework/blob/5.5/src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php).

I tried `round($value, 2)` before inserting to the database but it still fails, I'm guessing either 1.61 or 1.6071 can't be represented in binary accurately. I also tried adding `ini_set('precision', 17);` inside my `/** @test */` function but it still fails. I could probably just change the migration file to `$table->decimal('field', 10, 2)` and save myself all this trouble but unfortunately I can't because of project restrictions.

Anyway, I'm not going to argue if this is or isn't an issue with the framework as my knowledge is currently limited. However, if anyone found a way to use `assertDatabaseHas` with floats/doubles that would be great, my current google search shows no feasible solutions. My last resort would be to use `assertEquals` with $delta.
